### DISCUSSION TITLE
cross-compilation with hatch, integration in existing python project

### DISCUSSION THREAD

**Discussion by adlotsof** [PyO3/maturin#1968] [Q&A]
I got an existing python project and i would like to use as much as the existing build system (hatch and hatchling) as possible, but also include an rust extension module.
I manage to build using `hatch build` and `[tool.maturin]`, i am not sure how to get features like `post-install-commands` to work - but i also dont find any reference for how to build wheels for multiple architectures. I dont find any way to pass arguments to maturin from `hatch build` either. 
Is there some undocumented configuration for `maturin pep517` or is this a limtation of hatch?

**Comment by messense** [comment]
You can pass extra arguments to `maturin pep517 build-wheel` using `MATURIN_PEP517_ARGS` env var or the `build-arg` PEP 517 config settings.
